src/msg/async/frames_v2.cc

```diff
@@ -440,21 +454,23 @@ void FrameAssembler::disassemble_first_segment(bufferlist& preamble_bl,
 bool FrameAssembler::disassemble_remaining_segments(
     bufferlist segment_bls[], bufferlist& epilogue_bl) const {
   ceph_assert(!m_descs.empty());
+  bool is_completed = true;
   if (m_is_rev1) {
     if (m_descs.size() == 1) {
       // no epilogue if only one segment
       ceph_assert(epilogue_bl.length() == 0);
-      return true;
-    }
-    if (m_crypto->rx) {
-      return disasm_remaining_secure_rev1(segment_bls, epilogue_bl);
+    } else if (m_crypto->rx) {
+      is_completed = disasm_remaining_secure_rev1(segment_bls, epilogue_bl);
+    } else {
+      is_completed = disasm_remaining_crc_rev1(segment_bls, epilogue_bl);
     }
-    return disasm_remaining_crc_rev1(segment_bls, epilogue_bl);
-  }
-  if (m_crypto->rx) {
-    return disasm_all_secure_rev0(segment_bls, epilogue_bl);
+  } else if (m_crypto->rx) {
+    is_completed = disasm_all_secure_rev0(segment_bls, epilogue_bl);
+  } else {
+    is_completed = disasm_all_crc_rev0(segment_bls, epilogue_bl);
   }
-  return disasm_all_crc_rev0(segment_bls, epilogue_bl);
+
+  return is_completed;
```

Comment:
Have I understood correctly the entire change in `FrameAssembler::disassemble_remaining_segments()` is just a refactoring?

Reply:
Yes, it was a minor refactoring - but following your question I gave this code another look and changed it completly so that the segment disassembling would be entirely encapsulated within the assembler